### ISSUE TITLE
Profiling FeedingPatternInfant::Feeding container

### ISSUE THREAD

**Issue by pieter-edelman-nictiz** [Nictiz/Nictiz-R4-zib2020#353]
Zib FeedingPatternInfant::Feeding contains the container Feeding, which hosts two concepts: FeedingType (what is being fed) and FeedingMethod (how it is being fed). Both concepts have a ValueSet with two values, making for a total of four of which one is invalid (formula milk can only be fed using a bottle).

A problem arises when translating this model to FHIR on the Observation resource. Modelling additional information about an Observation is usually done using `Observation.component`. But this is very hard to achieve in this case, because these two concepts are dependent on each other, while at the same time the container can be used more than once (eg. infant is breast fed but also formula bottle fed). Defining `.component`s would break the dependency.

There are some possible solutions to this problem:
1. In the STU3 version there is a `.component` defined for the FeedingType concept, which is extended with the FeedingMethod concept. This works, but requires a custom extension.
2. The Feeding container could become a separate Observation profile with two `.component`s. This has the downside of breaking out part of the data model to a separate (set of) resource(s), only to solve a modelling problem.
3. The two concepts could be modeled as `.component`s with the requirement that their order dictates their relationship. One ore more constraints and `.orderMeaning` can be used to document/enforce this model. The downside is that this is "grouping by convention" which is not so intuitive, leading to extra documentation and possibly implementation errors.
4. Create a `.component` somehow containing the combined result of the two concepts -- after all, there are only three possible combinations. However, both ValueSets are extensible and it is not inconceivable that some use case introduces other concepts.

**Comment by pieter-edelman-nictiz** [comment]
See also https://bits.nictiz.nl/browse/GZ-779

**Comment by LilianMinne** [comment]
In my opinion it should be either option 1 or 2. Option 3 and 4 are a bit tricky, as you mention option 3 is more prone to errors and option 4 is not scalable. In our birthcare model we already define a third option for FeedingType (being donor milk) so you would already have 5 possible combinations. 

Note that we also asked the zib team some questions about this data model ([ZIB-1919](https://bits.nictiz.nl/browse/ZIB-1919)) as we wish to add more information to the Feeding container like quantity and duration of the feeding. We also need the FeedingFrequency to be linked to the Feeding container. For each Feeding (e.g. a baby might receive breast feeding as primary feeding and bottle milk as supplementary feeding) you should be able to specify a separate quantity, duration and frequency. This might even apply to FeedingSupplement, although a supplement can clearly only be added to a bottle and not when the patient is brest fed. 

The more concepts are linked to the Feeding container in the zib, the more I tend to vote for option 2. The Zib FeedingPatternInfant than only becomes a bundle of one or more Feeding Observations with an additional registry date and optional comment. I'm generally not too happy about breaking the data model in a seperate set of resources either, but we do it more often (e.g. zib Pregnancy) when it makes more sense in FHIR. Translations from zib data models to FHIR resources are not always one-to-one.

I can also live with option 1. As you mention, it works. The downside is that the more concepts you wish to link to the Feeding container, the more extensions you need.

**Comment by pieter-edelman-nictiz** [comment]
After some more discussion, option 2 seems to be the most elegant. However, the question arises how the Feeding Observation is linked to the main Observation. `Observation.hasMember` probably fits the bill, although the "normal" use cases are a bit different. Some further research needs to be done.

**Comment by pieter-edelman-nictiz** [comment]
So ... maybe it's an option to model this as a sequence of Observations which are linked using the Observation-sequalTo extension?

**Comment by LilianMinne** [comment]
I'm not sure if we should use the Observation-sequalTo extension. It implies there is a sequence in the Observations, which is not there in our use case?

> This observation follows the target observation (e.g. timed tests such as Glucose Tolerance Test).

Neither does it solve the problem, since I think these sequential Observations are also Observations that can be interpreted on their own. At least they will have different `.effective[x]` e.g. a Glucose Tolerance Test after 1-2-3 hours.

Another solution I thought of is using component together with `.valueReference`. Unfortunately Observation only supports `.component.valueReference` in the current build with a restriction to resource type MolecularSequence. So we would need an extension for that. It feels a bit like we are defining a custom way of grouping Observations while there is already a construction in place. Actually the fact that you create an Observation instance for each Feeding container already implies that these instances can be interpreted on their own. Bringing us back to the original discussion :(

By the way I wouldn't interpret `.effectiveDate` as the time of feeding, but rather the time the Observation was made about the feeding pattern.

Maybe we should pose a question on zulip about this?

**Comment by pieter-edelman-nictiz** [comment]
You're right, sequelTo doesn't work. But the same goes for `.valueReference` I think; the question remains if the Feeding container can be used as a separate Observation, and I think this is not the case (even if you would add all concepts from the zib) because you're missing information about the other Feeding containers.

You could perhaps approach this is a panel type Observation if the zib would define FeedingFrequency and FeedingSupplement in the Feeding container. Then you could make Feeding Observations that stand on their own, and a grouping Observation that defines a general Comment and a reference to zib NutritionAdvice. But for now this doesn't help.

I'll ask the question on Zulip.

**Comment by pieter-edelman-nictiz** [comment]
Zulip thread: https://chat.fhir.org/#narrow/stream/179166-implementers/topic/Profiling.20a.20.22multi-part.22.20component

**Comment by pieter-edelman-nictiz** [comment]
Zulip doesn't provide any new insights. We discussed this issue again with the profiling team and came to the conclusion that modelling the Feeding container as a separate Observation probably isn't quite right, and the that the zib model has issues that need to be resolved. In this light, it seems better to keep at the STU3 solution than to forge ahead with another approach that's not better in itself and needs to be changed again for the next zib release.

So in conclusion: we will continue with option 1.
